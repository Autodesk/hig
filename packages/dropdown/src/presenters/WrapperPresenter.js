import React from "react";
import PropTypes from "prop-types";
import { cx, css } from "emotion";

import stylesheet from "./WrapperPresenter.stylesheet";

/**
 * @typedef {Object} WrapperProps
 * @property {JSX.Element} [input]
 * @property {JSX.Element} [menu]
 * @property {boolean} [isOpen]
 */

/**
 * Downshift requires a reference to the component's root element.
 * As a result, this presenter should be purely functional so that it can
 * be used as function with Downshift.
 *
 * @param {WrapperProps} prop
 * @returns {JSX.Element}
 * @see https://github.com/paypal/downshift#getrootprops
 */
export default function WrapperPresenter(props) {
  const { disabled, children, ...otherProps } = props;
  const { className } = otherProps;

  return (
    <div {...otherProps} className={cx(css(stylesheet(props)), className)}>
      {children}
    </div>
  );
}

WrapperPresenter.propTypes = {
  children: PropTypes.node
};
