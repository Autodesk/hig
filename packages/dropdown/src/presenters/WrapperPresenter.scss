.hig__dropdown {
  position: relative;
  min-width: 300px;
  max-width: 450px;
}
