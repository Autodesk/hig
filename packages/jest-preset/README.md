# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
