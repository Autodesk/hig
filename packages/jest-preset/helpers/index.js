/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
