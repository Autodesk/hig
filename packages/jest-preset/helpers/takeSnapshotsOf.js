const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
