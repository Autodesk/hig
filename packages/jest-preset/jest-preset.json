{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
