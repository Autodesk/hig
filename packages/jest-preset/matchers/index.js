require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
