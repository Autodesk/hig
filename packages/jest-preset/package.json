{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
