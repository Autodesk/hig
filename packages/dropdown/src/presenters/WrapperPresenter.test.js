import React from "react";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

import WrapperPresenter from "./WrapperPresenter";

describe("Dropdown/presenters/WrapperPresenter", () => {
  const rendererOptions = {
    createNodeMock(element) {
      return document.createElement(element.type);
    }
  };
  const cases = [
    {
      desc: "renders without props",
      props: {}
    },
    {
      desc: "renders with all props",
      props: {
        disabled: true,
        children: <p key="p">hi</p>
      }
    }
  ];

  takeSnapshotsOf(WrapperPresenter, cases, rendererOptions);
});
