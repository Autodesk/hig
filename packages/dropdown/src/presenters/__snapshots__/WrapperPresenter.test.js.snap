// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown/presenters/WrapperPresenter renders with all props 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
}

<div
  className="emotion-0"
>
  <p>
    hi
  </p>
</div>
`;

exports[`Dropdown/presenters/WrapperPresenter renders without props 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
}

<div
  className="emotion-0"
/>
`;
